src/Data/ProgrammesDb/EntityRepository/ServiceRepository.php

```diff
@@ -20,16 +20,23 @@ public function findByIds(array $ids): array
             ->getQuery()->getResult(Query::HYDRATE_ARRAY);
     }
 
-    public function findByIdsWithNetworkServicesList(array $ids): array
+    public function findByIdsWithNetworkServicesList(array $ids, ?DateTimeImmutable $date = null): array
     {
-        return $this->createQueryBuilder('service')
+        $qb = $this->createQueryBuilder('service')
             ->addSelect(['masterBrand', 'network', 'networkServices'])
             ->leftJoin('service.masterBrand', 'masterBrand')
             ->leftJoin('service.network', 'network')
             ->leftJoin('network.services', 'networkServices')
             ->andWhere("service.id IN (:ids)")
-            ->setParameter('ids', $ids)
-            ->getQuery()->getResult(Query::HYDRATE_ARRAY);
+            ->setParameter('ids', $ids);
+
+        if ($date) {
+            $qb->andWhere('(service.startDate IS NULL OR service.startDate <= :date)');
```

Comment:
Am I being stupid, or should this be networkServices.startDate etc?

Reply:
Nope, copy-paste fail